content/sensu-core/2.0/reference/tokens.md

```diff
@@ -0,0 +1,148 @@
+weight: 1
+version: "2.0"
+product: "Sensu Core"
+menu: 
+  sensu-core-2.0:
+    parent: reference
+---
+
+## How do tokens work?
+When a check is scheduled to be executed by an agent, it first goes through a token substitution step. Any tokens matching attribute values in the check are applied, and then the check is executed. Invalid templates or unmatched tokens will return an error, which is logged and sent to the Sensu backend message transport. Checks with token matching errors will not be executed.
+
+## New and improved tokens
+Sensu 2.0 uses the [Go template][1] package to implement token substitution. Instead of using triple colons `:::` as in [1.x token substitution][2], 2.0 token substitution uses double curly braces `{{` around tokens to be substituted.
+
+## Sensu tokens specification
+
+### Token substitution syntax
+
+Tokens are invoked by wrapping references to entity or custom attributes with double curly braces, such as `{{ .ID }}` to substitute an entity's ID value. Nested Sensu [entity attributes][3] can be accessed via dot notation (e.g. `System.arch`).
+
+- `{{ .ID }}` would be replaced with the [entity `ID` attribute][3]
+- `{{ .URL }}` would be replaced with a custom attribute called `url`
+- `{{ Disk.Warning }}` would be replaced with a custom attribute called
+  `warning` nested inside of a JSON hash called `disk`
+
+### Token substitution default values
+
+In the event that an attribute is not provided by the [entity][3], a token's default
+value will be substituted. Token default values are separated by a pipe character (`|`), and can be used to provide a "fallback value" for entities that are missing a specified token attribute.
+
+- `{{.URL | default "https://sensu.io"}}` would be replaced with a [custom  attribute][3] called `url`. If no such attribute called `url` is included in the client definition, the default (or fallback) value of `https://sensu.io` will be used to substitute the token.
+
+### Unmatched tokens
+
+If a token is unmatched during check preparation, the agent check handler will return an error, and the check will not be executed. Unmatched token errors will look similar to the following:
+
+{{< highlight shell >}}
+error: unmatched token: template: :1:22: executing "" at <.System.Hostname>: map has no entry for key "System"
+{{</ highlight >}}
+
+Check config token errors will be logged by the agent, and sent to Sensu backend message transport as a check failure.
+
+## Examples
+
+### Token substitution for check thresholds 
+
+In this example [check configuration][5], the `check-disk-usage.rb` command accepts `-w` (warning) and `-c` (critical)
+arguments to indicate the thresholds (as percentages) for creating warning or critical events. If no token substitutions are provided by a check configuration, it will use default values to create a warning event at 80% disk capacity (i.e. `{{ Disk.Warning | default 80 }}`), and a critical event at 90% capacity (i.e. `{{ Disk.Critical | default 90 }}`).
```

Comment:
The go template syntax requires a dot (`.`) before the first attribute, so instead of `Disk.Warning`, it should be `.Disk.Warning`. Same goes for `Disk.Critical`.